### PR TITLE
Add `sqidDecode()` function

### PR DESCRIPTION
**Pull request by rschu1ze**
Suggested here: https://github.com/orgs/sqids/discussions/18#discussioncomment-8022645

### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Added function `sqidDecode()` which decodes [Sqids](https://sqids.org/).

### CODE REVIEW

**Comment by robot-ch-test-poll3** [comment]
*This is an automated comment for commit f6ecf2b6174896aac0c2492f3df653e385936929 with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/58544/f6ecf2b6174896aac0c2492f3df653e385936929/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td><a href="https://github.com/ClickHouse/ClickHouse/blob/master/.github/PULL_REQUEST_TEMPLATE.md?plain=1">❌ failure</a></td></tr>
<tbody>
</table>

File: docs/en/sql-reference/functions/hash-functions.md [hunk 1]
```diff
@@ -1777,34 +1777,66 @@ Result:
...
+```
+
+## decodeSqid
+
+Decodes a [Sqid](https://sqids.org/) into numbers.
+
+**Syntax**
+
+```sql
+decodeSqid(number1, ...)
```
Comment (mgred) [inline docs/en/sql-reference/functions/hash-functions.md:1821]: It decodes from `number` to `string` :nerd_face:

Comment (rschu1ze) [reply docs/en/sql-reference/functions/hash-functions.md:1821]: Ah, of course ... fixed, thanks.

File: src/Functions/sqid.cpp [hunk 2]
```diff
@@ -80,21 +82,86 @@ class FunctionSqid : public IFunction
...
+        auto & res_nested_data = col_res_nested->getData();
+
+        auto col_res_offsets = ColumnArray::ColumnOffsets::create();
+        auto & res_offsets_data = col_res_offsets->getData();
+        res_offsets_data.reserve(input_rows_count);
+
+        for (size_t i = 0; i < input_rows_count; ++i)
+        {
+            const ColumnWithTypeAndName & arg = arguments[0];
+            ColumnPtr current_column = arg.column;
```
Comment (Avogar) [inline src/Functions/sqid.cpp:118]: We can move it outside the loop

File: src/Functions/sqid.cpp [hunk 3]
```diff
@@ -80,21 +82,86 @@ class FunctionSqid : public IFunction
...
+            std::string_view sqid = current_column->getDataAt(i).toView();
```
Comment (Avogar) [inline src/Functions/sqid.cpp:119]: Maybe better to use specific `ColumnString` here instead of calling virtual `getDataAt`? Or it's not a big deal?

Comment (rschu1ze) [reply src/Functions/sqid.cpp:119]: Sure, why not, at least the compiler can devirtualize then (`ColumnString` is marked `final`).

**Comment by rschu1ze** [comment]
@Avogar: approve? Thanks

**Review by Avogar — APPROVED** [review]

**Pull request merged**